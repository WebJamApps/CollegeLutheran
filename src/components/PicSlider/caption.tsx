import React from 'react';

const Caption = (props: { caption?: (JSX.Element|string); }):JSX.Element => {
  const { caption } = props;
  return (
    <div
      className="slider-caption"
      style={{
        textAlign: 'center',
        fontWeight: 'bold',
        padding: 0,
        backgroundColor: 'silver',
        marginBottom: 0,
        marginTop: 0,
      }}
    >
      {' '}
      {caption}
      {' '}
    </div>
  );
};

Caption.defaultProps = {
  caption: '',
};

export default Caption;
