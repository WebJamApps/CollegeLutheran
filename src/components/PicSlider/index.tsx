import React, { Component } from 'react';
import Slider from 'react-slick';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import Caption from './caption';
import { Ibook } from '../../redux/mapStoreToProps';

type SettingsType = {
  autoplay: boolean;
  autoplaySpeed: number;
  infinite: boolean;
  speed: number;
  slidesToShow: number;
  slidesToScroll: number;
  arrows: boolean;
  fade: boolean;
};

export interface PicSliderProps {
  settings?: SettingsType;
  data: Ibook[];
}
class PicSlider extends Component<PicSliderProps> {
  static defaultProps: { data: [{ url: ''; title: ''; _id: 0 }]; };

  settings: SettingsType;

  constructor(props: Readonly<PicSliderProps>) {
    super(props);
    this.settings = {
      autoplay: true,
      autoplaySpeed: 3000,
      infinite: true,
      speed: 5000,
      slidesToShow: 1,
      slidesToScroll: 1,
      arrows: false,
      fade: true,
    };
  }

  /* eslint-disable react/jsx-props-no-spreading */
  render(): JSX.Element {
    const { data } = this.props;
    return (
      <div>
        <Slider {...this.settings}>
          {
            data.map((d: Ibook) => (
              <div key={d._id}>
                <img className="slide-show" src={d.url} alt={d.title} />
                {d.comments === 'showCaption' ? <Caption caption={d.title} /> : null}
              </div>
            ))
          }
        </Slider>
      </div>
    );
  }
}

export default PicSlider;
