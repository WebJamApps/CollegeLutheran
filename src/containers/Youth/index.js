import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DefaultYouthContent from './YouthContent';
import mapStoreToProps from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';

export const Youth = ({ youthPics }) => {
  commonUtils.setTitleAndScroll('Youth Ministry');
  return (<DefaultYouthContent youthPics={youthPics} />);
};

Youth.defaultProps = { youthPics: [] };
Youth.propTypes = {
  youthPics: PropTypes.arrayOf(PropTypes.shape({
    comments: PropTypes.string,
  })),
};

export default connect(mapStoreToProps, null)(Youth);
