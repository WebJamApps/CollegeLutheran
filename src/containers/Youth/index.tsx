import React from 'react';
import { connect } from 'react-redux';
import DefaultYouthContent from './YouthContent';
import mapStoreToProps, { Ibook } from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';

export interface YouthProps {
  youthPics?: Ibook[];
}

export const Youth = ({ youthPics }: YouthProps): JSX.Element => {
  commonUtils.setTitleAndScroll('Youth Ministry', window.screen.width);
  return (<DefaultYouthContent youthPics={youthPics} />);
};

Youth.defaultProps = { youthPics: [] };

export default connect(mapStoreToProps, null)(Youth);
