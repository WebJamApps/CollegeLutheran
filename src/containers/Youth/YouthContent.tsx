import React from 'react';
import PicSlider from '../../components/PicSlider';
import type { YouthProps } from './index';

const YouthContent = ({ youthPics }: YouthProps): JSX.Element => (
  <div className="page-content">
    <div id="youthSlideshowWide" style={{ display: 'flex', flexDirection: 'column' }}>
      {youthPics && youthPics.length > 0 ? (<PicSlider data={youthPics} />) : null}
    </div>
    <div className="container-fluid">
      <p style={{ fontSize: '4pt', margin: '0' }}>&nbsp;</p>
      <div className="material-content elevation3" style={{ maxWidth: '998px', paddingBottom: '-80px', margin: 'auto' }}>
        <h3 style={{ paddingTop: '22px', paddingBottom: '15px' }}>Youth Ministry</h3>
        <p>
          <span style={{ color: 'rgb(0, 51, 102)' }}>
            <strong>
              <i>&quot;Think of us in this way, as servants of Christ and stewards of God’s mysteries.&quot;</i>
              &nbsp;&nbsp;&nbsp;1 Corinthians 4:1
            </strong>
          </span>
        </p>
        <p>
          Youth ministry (6th- 12th grades) at College Lutheran church focuses on God’s mysteries and serving Christ. We are deeply committed to
          helping students explore and live out their faith. Weekly Sunday school, worship, and youth group, as
          well as annual Virginia Synod youth events challenge our students to think deeply about their religious
          beliefs and build on the foundations of the Christian faith. These gatherings provide dynamic contexts for students to examine how
          real-world issues and religion are intertwined. We believe that when our students are encouraged to think about what they believe and ask
          questions, they develop a stronger faith that can withstand the complexities of the teen years and
          adulthood.
        </p>
        <p>
          <span style={{ color: 'rgb(0, 51, 102)' }}>
            <strong>
              <i>&quot;Faith without works is dead.&quot;</i>
              &nbsp;&nbsp;&nbsp;James 2:17
            </strong>
          </span>
        </p>
        <p>
          Not only do our youth have faith, but they show it at every opportunity! Youth at CLC are lifted up and celebrated as essential members
          in the worship, service, and recreational life of
          the congregation. They assist in worship as acolytes, crucifers, communion assistants, assisting ministers, ushers, greeters,
          coffee hour hosts, and children’s sermon speakers. They serve the community through events such as the CROP
          walk for hunger, Trick or Treat So Others May Eat, Trunk or Treating, Souper Bowl Sunday,
          and environmental initiatives. And, they are always excited about fun youth group events like snowtubing, bowling, and hiking!
          <br />
          <br />
          We invite you to come be part of our active, growing youth ministry program at any or all of our upcoming events.
        </p>
        <section style={{ textAlign: 'left', marginTop: '0' }} />
      </div>
      <div className="youthELCA">
        <a href="http://www.elca.org/" target="_blank" rel="noopener noreferrer">
          <img
            id="elcaLogo"
            alt="ELCA LOGO"
            src="https://dl.dropboxusercontent.com/s/wkzubcmmm3pqst4/elca-logo.png?dl=0"
            style={{ width: '350px', paddingTop: '30px', margin: 'auto' }}
          />
        </a>
      </div>
      <p>&nbsp;</p>
    </div>
  </div>
);

YouthContent.defaultProps = { youthPics: [] };

export default YouthContent;
