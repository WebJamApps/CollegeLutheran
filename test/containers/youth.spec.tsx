/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Youth } from '../../src/containers/Youth/index';
import DefaultYouthContent from '../../src/containers/Youth/YouthContent';
import PicSlider from '../../src/components/PicSlider';

function setup(data: any[]) {
  let wrapper: any;
  if (data !== null && data !== undefined) {
    wrapper = shallow(<Youth youthPics={data} />);
  } else wrapper = shallow(<Youth />);
  return { wrapper };
}

describe('Youth', () => {
  it('renders correctly without images', () => {
    const tree = renderer
      .create(<Youth youthPics={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Renders the Youth component', () => {
    const data = [{}];
    const { wrapper } = setup(data);
    expect(wrapper.find(DefaultYouthContent).exists()).toBe(true);
    expect(wrapper.find(DefaultYouthContent).dive().find('div.page-content').exists()).toBe(true);
  });
  it('renders with images', () => new Promise((done) => {
    const data:any[] = [{
      title: '', type: '', _id: '', created_at: '',
    }];
    const wrapper2 = shallow(<DefaultYouthContent youthPics={data} />);
    expect(wrapper2.find(PicSlider).exists()).toBe(true);
    done();
  }));
});
