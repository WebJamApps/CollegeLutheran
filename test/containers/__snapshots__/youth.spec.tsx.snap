// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Youth renders correctly without images 1`] = `
<div
  className="page-content"
>
  <div
    id="youthSlideshowWide"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  />
  <div
    className="container-fluid"
  >
    <p
      style={
        Object {
          "fontSize": "4pt",
          "margin": "0",
        }
      }
    >
       
    </p>
    <div
      className="material-content elevation3"
      style={
        Object {
          "margin": "auto",
          "maxWidth": "998px",
          "paddingBottom": "-80px",
        }
      }
    >
      <h3
        style={
          Object {
            "paddingBottom": "15px",
            "paddingTop": "22px",
          }
        }
      >
        Youth Ministry
      </h3>
      <p>
        <span
          style={
            Object {
              "color": "rgb(0, 51, 102)",
            }
          }
        >
          <strong>
            <i>
              "Think of us in this way, as servants of Christ and stewards of God’s mysteries."
            </i>
               1 Corinthians 4:1
          </strong>
        </span>
      </p>
      <p>
        Youth ministry (6th- 12th grades) at College Lutheran church focuses on God’s mysteries and serving Christ. We are deeply committed to helping students explore and live out their faith. Weekly Sunday school, worship, and youth group, as well as annual Virginia Synod youth events challenge our students to think deeply about their religious beliefs and build on the foundations of the Christian faith. These gatherings provide dynamic contexts for students to examine how real-world issues and religion are intertwined. We believe that when our students are encouraged to think about what they believe and ask questions, they develop a stronger faith that can withstand the complexities of the teen years and adulthood.
      </p>
      <p>
        <span
          style={
            Object {
              "color": "rgb(0, 51, 102)",
            }
          }
        >
          <strong>
            <i>
              "Faith without works is dead."
            </i>
               James 2:17
          </strong>
        </span>
      </p>
      <p>
        Not only do our youth have faith, but they show it at every opportunity! Youth at CLC are lifted up and celebrated as essential members in the worship, service, and recreational life of the congregation. They assist in worship as acolytes, crucifers, communion assistants, assisting ministers, ushers, greeters, coffee hour hosts, and children’s sermon speakers. They serve the community through events such as the CROP walk for hunger, Trick or Treat So Others May Eat, Trunk or Treating, Souper Bowl Sunday, and environmental initiatives. And, they are always excited about fun youth group events like snowtubing, bowling, and hiking!
        <br />
        <br />
        We invite you to come be part of our active, growing youth ministry program at any or all of our upcoming events.
      </p>
      <section
        style={
          Object {
            "marginTop": "0",
            "textAlign": "left",
          }
        }
      />
    </div>
    <div
      className="youthELCA"
    >
      <a
        href="http://www.elca.org/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="ELCA LOGO"
          id="elcaLogo"
          src="https://dl.dropboxusercontent.com/s/wkzubcmmm3pqst4/elca-logo.png?dl=0"
          style={
            Object {
              "margin": "auto",
              "paddingTop": "30px",
              "width": "350px",
            }
          }
        />
      </a>
    </div>
    <p>
       
    </p>
  </div>
</div>
`;
